src/OpalCompiler-Core/TemporaryVariable.class.st

```diff
@@ -13,6 +13,11 @@ TemporaryVariable class >> semanticNodeClass [
 	^RBTemporaryNode 
 ]
 
+{ #category : #visiting }
+TemporaryVariable >> acceptVisitor: aProgramNodeVisitor node: aNode [
+	^ aProgramNodeVisitor visitTemporaryNode: aNode
```

Comment:
We should also cleanup methods with temp in name to have localVariable meaning. Here it would be #visitLocalVariableNode:.

Reply:
There is some real problem in the AST... #visitArgumentNode: is called for *declarations*, while visitTemporaryNode: is how you would expect it. There is #visitTemporaryDeclarationNode: for the temp declaration... which falls back to call #visitArgumentNode, too.

This requires quire some non-backward compatible refactoring to get right....